File: codegen/codegen-erlang/src/test/resources/ir/http_checksum_request_headers.expected.erl
Checksum0 = crypto:hash(md5, Body),
HeadersWithChecksum = headers_set(<<"Content-MD5">>, checksum_header_encode(Checksum0), Headers)

HeadersWithChecksum = case ChecksumAlgorithm of
    undefined -> Headers;
    crc32c ->
        Checksum = crc32c_hash(Body),
        headers_set(<<"x-amz-checksum-crc32c">>, checksum_header_encode(Checksum), Headers);
    sha256 ->
        Checksum = crypto:hash(sha256, Body),
        headers_set(<<"x-amz-checksum-sha256">>, checksum_header_encode(Checksum), Headers);
    Other -> error({unsupported_checksum_algorithm, Other})
end

File: codegen/codegen-elixir/src/test/resources/ir/http_checksum_request_headers.expected.ex
checksum0 = :crypto.hash(:md5, body)
headers = HttpChecksum.headers_set("Content-MD5", HttpChecksum.checksum_header_encode(checksum0), headers)

headers = case input.checksum_algorithm do
  nil -> headers
  :crc32c ->
    checksum = HttpChecksum.crc32c_hash(body)
    HttpChecksum.headers_set("x-amz-checksum-crc32c", HttpChecksum.checksum_header_encode(checksum), headers)
  :sha256 ->
    checksum = :crypto.hash(:sha256, body)
    HttpChecksum.headers_set("x-amz-checksum-sha256", HttpChecksum.checksum_header_encode(checksum), headers)
  other -> Kernel.raise(:ArgumentError, {:unsupported_checksum_algorithm, other})
end
